### PR TITLE
Bump dependencies for when still running on ruby 2.7

### PR DESCRIPTION
**Pull request by faisal**
Bump cucumber, minitest, parser, rake, reek, rubocop, and ruby_parser dependencies.

This is a pull request to update dependencies, intended for landing as part of an update that we put together for other reasons. I expect to update it from time to time when I notice other relevant dependency updates.

Check list:
- [X] Add an entry to the [changelog](https://github.com/whitesmith/rubycritic/blob/main/CHANGELOG.md)
- [X] [Squash all commits into a single one](https://github.com/whitesmith/rubycritic/blob/main/CONTRIBUTING.md)
- [X] Describe your PR, link issues, etc.

### CODE REVIEW

**Review by etagwerker — APPROVED** [review]
@faisal This looks good, thanks! Before I merge it, I'd like to know what you think about https://github.com/whitesmith/rubycritic/issues/482

We might want to start requiring Ruby 3.0 or higher in order to have more current core dependencies (e.g. reek)

**Comment by faisal** [comment]
I'll comment more fully there, but for purposes of this PR I do think that supporting Ruby 2.7 is starting to hold us back. For example, I believe that updating the minimum Ruby to 3.0 would allow currently us to pull our dependencies up to minitest 5.22.2, rake 13.1.0, and rubocop 1.61.0.

**Comment by faisal** [comment]
I started [a branch for dropping 2.7 support](https://github.com/faisal/rubycritic/tree/ruby_3). Once we know what we're doing with _this_ PR, it should be a quick turn to make a PR out of that branch.

**Comment by faisal** [comment]
Is it worth integrating the fix from https://github.com/whitesmith/rubycritic/pull/486 ?

**Review by nunosilva800 — APPROVED** [review]
Since Reek 6.2+ requires ruby 3, we can make RubyCritic v4.9.x the last one with ruby 2.7 support. 

We can either require a min of ruby v3.0.0 in RubyCritic v4.10, or make it v5.0. Don't really have a strong opinion on this.

File: rubycritic.gemspec [hunk 1]
```diff
@@ -33,11 +33,11 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency 'flay', '~> 2.13'
   spec.add_runtime_dependency 'flog', '~> 4.7'
   spec.add_runtime_dependency 'launchy', '>= 2.5.2'
-  spec.add_runtime_dependency 'parser', '>= 3.2.2.1'
+  spec.add_runtime_dependency 'parser', '>= 3.2.2.4'
   spec.add_runtime_dependency 'rainbow', '~> 3.1.1'
-  spec.add_runtime_dependency 'reek', '~> 6.0', '< 7.0'
+  spec.add_runtime_dependency 'reek', '~> 6.1', '< 7.0' # >= 6.2 requires Ruby 3
```
Comment (nunosilva800) [inline rubycritic.gemspec:38]:
```suggestion
  spec.add_runtime_dependency 'reek', '~> 6.0', '< 6.2'
```

**Pull request merged**